busbus/provider/gtfs.py

```diff
@@ -47,23 +72,136 @@ def __init__(self, provider, **data):
+        data['service'] = provider.get(GTFSService, data['_service_id'])
+        if '_block_id' in data:
+            pass  # FIXME
+        if '_shape_id' in data:
+            pass  # FIXME
+        if '_accessible' in data:
+            pass  # FIXME
+        if '_bikes' in data:
+            data['bikes_ok'] = {u'0': None, u'1': True,
+                                u'2': False}[data['_bikes']]
+        super(GTFSTrip, self).__init__(provider, **data)
+
+    @property
+    def frequencies(self):
+        return Queryable(sorted(
+            self._provider._get_relation(self, 'frequencies'),
+            key=operator.attrgetter('start_time')))
+
+    @property
+    def stop_times(self):
+        return Queryable(sorted(
+            self._provider._get_relation(self, 'stop_times'),
+            key=operator.attrgetter('sequence')))
+
+
+class GTFSTripFrequency(busbus.entity.BaseEntity):
+    __attrs__ = ('trip', 'start_time', 'end_time', 'headway', 'exact_times')
+    __repr_attrs__ = ('trip', 'start_time', 'end_time')
+
+    def __init__(self, provider, **data):
+        data['trip'] = provider.get(GTFSTrip, data['_trip_id'])
+        for attr in ('start_time', 'end_time'):
+            data[attr] = parse_gtfs_time(data[attr])
+        data['headway'] = int(data['headway'])
+        if 'exact_times' in data:
+            data['exact_times'] = bool(int(data['exact_times']))
+        provider._add_relation(GTFSTrip, data['_trip_id'], 'frequencies', self)
+        super(GTFSTripFrequency, self).__init__(provider, **data)
+
+
+class GTFSStopTime(busbus.entity.BaseEntity):
+    __attrs__ = ('trip', 'stop', 'arrival_time', 'departure_time', 'sequence',
+                 'headsign', 'pickup', 'dropoff', 'shape_dist_traveled',
+                 'exact_times')
+
+    def __init__(self, provider, **data):
+        data['trip'] = provider.get(GTFSTrip, data['_trip_id'])
+        data['stop'] = provider.get(busbus.Stop, data['_stop_id'])
+        for attr in ('arrival_time', 'departure_time'):
+            if attr in data:
+                data[attr] = parse_gtfs_time(data[attr])
+        if 'departure_time' in data:
+            if data['departure_time'] == data['arrival_time']:
+                del data['departure_time']
+        data['sequence'] = int(data['sequence'])
+        if 'pickup' in data:
+            pass  # FIXME
+        if 'dropoff' in data:
+            pass  # FIXME
+        if 'shape_dist_traveled' in data:
+            data['shape_dist_traveled'] = float(data['shape_dist_traveled'])
+        if 'exact_times' in data:
+            data['exact_times'] = {u'0': False, u'1': True}.get(
+                data['exact_times'], True)
+        provider._add_relation(GTFSTrip, data['_trip_id'], 'stop_times', self)
+        provider._add_relation(busbus.Stop, data['_stop_id'], 'trips',
+                               data['trip'])
+        super(GTFSStopTime, self).__init__(provider, **data)
+
+
+GTFS_FILENAME_MAP = OrderedDict([
```

Comment:
This large ordered dictionary is good for having the entire key mapping in one place. An alternate approach based around reading `__attrs__` and, optionally, designated method(s) for remapping GTFS column names to Python property names, could reduce the "big blob of data" factor here.